volatility/framework/plugins/windows/poolscanner.py

```diff
@@ -91,6 +91,10 @@ def __call__(self, data: bytes, data_offset: int):
                         checks_pass = True
                     elif (constraint.page_type & PoolType.NONPAGED) and header.PoolType % 2 == 1:
                         checks_pass = True
+                    elif (constraint.page_type
```

Comment:
@iMHLv2, can you please give this a quick look over?  It's a change to the core so I want to make sure you're happy with it...

Reply:
(Or @superponible, just someone windows-y to ok this as a core change).  5:)

Reply:
@ikelos @japhlange @iMHLv2 This doesn't look necessary but there's actually a different problem.  All the builtin constraints check paged, nonpaged, and free, so it wasn't evident. The page type checks in vol2's poolscan.py use the FreePool, NonPagedPool, and PagedPool properties defined in _POOL_HEADER here: https://github.com/volatilityfoundation/volatility/blob/master/volatility/plugins/overlays/windows/windows.py#L1153 

It looks like vol3's PoolHeaderScanner took those checks directly into the page type checks here: https://github.com/volatilityfoundation/volatility3/blob/master/volatility/framework/plugins/windows/poolscanner.py#L88

However, vol2 redefines those checks in https://github.com/volatilityfoundation/volatility/blob/master/volatility/plugins/overlays/windows/vista.py#L42, basically swapping PagedPool and NonPagedPool.

So the vol3 page type checks will need to be modified to use a different check based on the OS version.

Reply:
Hmmm, so I think @superponible is gonna look into this and we may make some core changes that will simplify (but require changes to) this merge request.  Nearly there!  5:)

Reply:
Give me a shout if anything is needed or if I can help in any way.